rdmo/views/managers.py

```diff
@@ -10,16 +10,26 @@
 )
 
 
-class ViewQuestionSet(CurrentSiteQuerySetMixin, GroupsQuerySetMixin, AvailabilityQuerySetMixin, models.QuerySet):
+class ViewQuestionSet(CurrentSiteQuerySetMixin, GroupsQuerySetMixin, AvailabilityQuerySetMixin, QuerySet):
 
     def filter_catalog(self, catalog):
-        return self.filter(models.Q(catalogs=None) | models.Q(catalogs=catalog))
+        return self.filter(Q(catalogs=None) | Q(catalogs=catalog))
 
+    def filter_available_views_for_project(self, project):
+        site_filter = Q(sites=project.site) | Q(sites__isnull=True)
+        catalogs_filter = Q(catalogs=project.catalog) | Q(catalogs__isnull=True)
+        groups_filter = Q(groups__in=project.groups) | Q(groups__isnull=True)
+        availability_filter = Q(available=True)
+        return self.filter(site_filter & catalogs_filter & groups_filter & availability_filter)
```

Comment:
I am sorry but there is a newline in `rdmo.tasks.managers` right here :upside_down_face: